contracts/FIFOMarket.sol

```diff
@@ -24,6 +26,8 @@ contract FIFOMarket is
   ERC1155HolderUpgradeable,
   IERC777RecipientUpgradeable
 {
+  using RemovalUtils for *;
```

Comment:
What's the goal of the using for '*' here?  My understanding from the docs:

https://docs.soliditylang.org/en/v0.8.13/contracts.html#using-for

Is that this would attach the methods of `RemovalUtils` to all types in scope.  I think given the way you're calling those functions in a qualified way (i.e. `RemovalUtils.supplierAddressFromRemovalId(_queue[i])`) that this line may not be needed.

Reply:
I think there is a path to using using :) that might look something like:

```suggestion
  using RemovalUtils for uint256;

  function test() {
    uint256 myId = 12345;
    console.log(myId.supplierAddress());
  }
```

Where the method still takes the id as the first argument but the binding (this is very similar to `.bind()` in javascript to set `this`).

```
  function supplierAddress(uint256 removalId)
    internal
    pure
    returns (address)
  {
    return
      address(
        uint160(
          _extractValue(removalId, _ADDRESS_FIELD_LENGTH, _ADDRESS_OFFSET)
        )
      );
  }
```